src/cases/containers/case-hearings/case-hearings.component.ts

```diff
@@ -39,8 +37,35 @@ export class CaseHearingsComponent implements OnInit, OnDestroy {
   }
 
   public ngOnInit(): void {
-    this.upcomingHearings$ = this.getHearsListByStatus(EXUISectionStatusEnum.UPCOMING);
-    this.pastAndCancelledHearings$ = this.getHearsListByStatus(EXUISectionStatusEnum.PAST_AND_CANCELLED);
+    this.getHearsListByStatus(EXUISectionStatusEnum.UPCOMING).subscribe(hearings => {
+      if (hearings.length) {
+        const viewModels: CaseHearingViewModel[] = this.convertServiceToVM(hearings);
+
+        this.upcomingHearings$ = of(viewModels.sort((a, b) => {
+          return new Date(a.hearingRequestDateTime) > new Date(b.hearingRequestDateTime) ? 1 : -1;
+        }).sort((a, b) => {
+          return new Date(a.mostRecentHearingStartDateTime) > new Date(b.mostRecentHearingStartDateTime) ? 1 : -1;
+        }).sort((a) => {
+          return a.hearingListingStatus === HearingListingStatusEnum.AWAITING_LISTING ? -1 : 1;
+        }));
```

Comment:
according to requirement, should we order the mostRecentHearingStartDateTime first, if  mostRecentHearingStartDateTime  is not found(null) then order by hearingRequestDateTime?

Reply:
This done the sort chaining is reverse order